trieste/models/interfaces.py

```diff
@@ -250,7 +250,7 @@ def conditional_predict_y(
         )
 
 
-class ModelStack(Generic[T], ProbabilisticModel):
+class ModelStack(ProbabilisticModel, Generic[T]):
```

Comment:
Is that it? Nice fix! Can you explain a bit why the order matters here?

Reply:
No idea! But it seems to work (and doesn't break anything on 3.7). It's probably some weird interaction between Protocol and Generic.

Reply:
Looks like Protocol (unlike ABC) inherits from Generic, meaning we need to put it first to ensure monotonicity ("a class always appears before its ancestor"). See https://en.wikipedia.org/wiki/C3_linearization for more details if you're interested.

Reply:
Thanks for the reference! Apparently this wasn't the case in 3.7, hence no error. Also, worth noting that `Protocol` was moved from extensions to `typing` package in 3.8.

Reply:
Yes. We're only using typing_extensions because we support 3.7.